### PR TITLE
Feature: image fetcher

### PR DESCRIPTION
**Pull request by vadymmarkov**
@RamonGilabert @zenangst So it should fetch something.

### CODE REVIEW

File: Source/ImageFetcher.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  public enum Error: ErrorType {
+    case InvalidResponse
+    case InvalidStatusCode
+    case InvalidData
+    case ConvertionError
+  }
+
+  public let URL: NSURL
+  var task: NSURLSessionDataTask?
+  var active = true
```
Comment (zenangst) [inline Source/ImageFetcher.swift]: Should this be `false` as the default?

File: Source/ImageFetcher.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    task?.resume()
+  }
+
+  public func cancel() {
+    task?.cancel()
+    active = false
+  }
+
+  func complete(closure: () -> Void) {
```
Comment (zenangst) [inline Source/ImageFetcher.swift:76]: Should we set `active` to `false` here as well? Seems like we only do that in `cancel()`

**Comment by vadymmarkov** [comment]
@zenangst Should be fixed now :wink:

**Comment by zenangst** [comment]
@vadymmarkov :tada:

**Pull request merged**